Tensile/KernelWriterAssembly.py

```diff
@@ -5211,30 +5208,237 @@ def lraFinalOffset(self, kernel, tP):
           "sgid=sgid*(MT%u+PAD)"%tIdx )
 
       # TODO: generalize over different MIs
-      if kernel["MatrixInstruction"] and not kernel["ProblemType"]["DataType"].isHalf():
-        if "MatrixInstK" in kernel and kernel["MatrixInstK"] > 1 and tP["bpe"] == 4:
-          kDiv = kernel["MatrixInstN"]
-          kStr += vectorStaticDivide(rReg, dividendReg, kDiv, tmpVgpr, tmpSgpr)
-          kStr += inst("v_and_b32", \
-              vgpr(rReg), \
-              hex(kernel["MatrixInstK"]-1), \
-              vgpr(rReg), \
-              "k groups")
-          kStr += inst("s_mov_b32", \
-              sgpr(tmpSgpr), \
-              hex(kernel["MacroTile%u"%tIdx] + kernel["LdsPad%s"%tc]), \
-              "MT%u+PAD"%tIdx )
-          kStr += inst("v_mul_lo_u32", \
-              vgpr(rReg), \
-              sgpr(tmpSgpr), \
-              vgpr(rReg), \
-              "koff")
-          kStr += inst("v_add_u32", vgpr(sgid), vgpr(rReg), vgpr(sgid), "k offset")
+      #if kernel["MatrixInstruction"] and not kernel["ProblemType"]["DataType"].isHalf():
+      #  if "MatrixInstK" in kernel and kernel["MatrixInstK"] > 1 and tP["bpe"] == 4:
+      #    kDiv = kernel["MatrixInstN"]
+      #    kStr += vectorStaticDivide(rReg, dividendReg, kDiv, tmpVgpr, tmpSgpr)
+      #    kStr += inst("v_and_b32", \
+      #        vgpr(rReg), \
+      #        hex(kernel["MatrixInstK"]-1), \
+      #        vgpr(rReg), \
+      #        "k groups")
+      #    kStr += inst("s_mov_b32", \
+      #        sgpr(tmpSgpr), \
+      #        hex(kernel["MacroTile%u"%tIdx] + kernel["LdsPad%s"%tc]), \
+      #        "MT%u+PAD"%tIdx )
+      #    kStr += inst("v_mul_lo_u32", \
+      #        vgpr(rReg), \
+      #        sgpr(tmpSgpr), \
+      ##        vgpr(rReg), \
+      #        "koff")
+      #    kStr += inst("v_add_u32", vgpr(sgid), vgpr(rReg), vgpr(sgid), "k offset")
       # end TODO
       if kernel["VectorWidth"] > 1 and not kernel["MatrixInstruction"]:
         kStr += staticMultiply(vgpr(tP["gpr"]["lro"]), 
             vgpr(tP["gpr"]["lro"]), \
             kernel["VectorWidth"], sgpr(tmpSgpr))
+
+      #if kernel["MatrixInstruction"] and not kernel["ProblemType"]["DataType"].isHalf():
+      # miWG0 requiriing > 1 SIMD case
+      if kernel["MatrixInstruction"]:
+        if tc == "A":  
+            #sub-case 1: MatrixInstB != case 
```

Comment:
Should there be sub-case 2 for other sizes?

Reply:
subcase is below numBlocks=1